include/core/simulation_control.h

```diff
@@ -505,6 +505,12 @@ class SimulationControl
    */
   std::vector<double>
   get_checkpointed_simulation_control_info(const std::string &prefix);
+
+  void
+  increment_iteration()
+  {
+    iteration_number++;
+  };
```

Comment:
I don't think that's a good idea. I would instead use the simulation control we already have for DEM instead or an existing simulation control instead of adding a new way to "move the iteration forward"
Worst case make a SimulationControlSteps or something simiilar, but the simulation control always work with a while(simulation_control->iterate()) and I would like to keep it this way. Otherwise one of the model is not iterated in the same way

Reply:
Yes or create another SimulationControl which just prints out steps. But the logic of using a while and a simulation control needs to be kinda universal. You can provide the stopping information to the simulation control objectif if need be.